src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/GZipEncoderDecoder/GZipEncoderDecoderToolProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿#nullable enable
+
+using System.Composition;
+using DevToys.Shared.Api.Core;
+using DevToys.Api.Tools;
+using DevToys.Core.Threading;
+using Windows.UI.Xaml.Controls;
+
+namespace DevToys.ViewModels.Tools.GZipEncoderDecoder
+{
+    [Export(typeof(IToolProvider))]
+    [Name("GZip Compress/Decompress")]
+    [Parent(EncodersDecodersGroupToolProvider.InternalName)]
+    [ProtocolName("gzip")]
+    [Order(1)]
+    internal sealed class GZipEncoderDecoderToolProvider : ToolProviderBase, IToolProvider
+    {
+        public string MenuDisplayName => LanguageManager.Instance.GZipEncoderDecoder.MenuDisplayName;
+
+        public string? SearchDisplayName => LanguageManager.Instance.GZipEncoderDecoder.SearchDisplayName;
+
+        public string? Description => LanguageManager.Instance.GZipEncoderDecoder.Description;
+
+        public string AccessibleName => LanguageManager.Instance.GZipEncoderDecoder.AccessibleName;
+
+        public TaskCompletionNotifier<IconElement> IconSource => CreateSvgIcon("Gzip.svg"); // TODO: icon. Dont know how to create svg stuff
```

Comment:
Nobody improvises being a designer. Look, @btiteux and I took more than a week to do the DevToys icon and now we realized we need to change it because of #238 😅